### PR TITLE
docs: Add DEVELOPMENT.md

### PR DESCRIPTION
**Pull request by aborgna-q**
Mostly copied from HUGR's.

drive-by: Remove "experimental execution" (`guppy-runner`) section from the readme.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+To setup the environment manually you will need:
+
+- Just: https://just.systems/
+- uv `>=0.3`: docs.astral.sh/uv/getting-started/installation
+
+The extended test suite has additional requirements. These are **optional**; tests that require them will be skipped if they are not installed.
+
+- Rust `>=1.75`: https://www.rust-lang.org/tools/install
+- `llvm-14`: https://apt.llvm.org/
```
Comment (qartik) [inline DEVELOPMENT.md:37]:
```suggestion
- `llvm-14`: https://llvm.org
```
`apt` is just for Debian-like linux distributions.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+- chore: Updating build tasks, package manager configs, etc. These changes are not published in the changelog.
+- revert: Reverting previous commits.
+
+## :shipit: Releasing new versions
+
+We use automation to bump the version number and generate changelog entries
+based on the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) labels. Release PRs are created automatically
+for each package when new changes are merged into the `main` branch. Once the PR is
+approved by someone in the [release team](.github/CODEOWNERS) and is merged, the new package
+is published on PyPI or crates.io as appropriate.
```
Comment (ss2165) [inline DEVELOPMENT.md:133]: remove crates reference

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+you can create a new branch from the latest release tag and cherry-pick the commits
+you want to include in the patch release.
+
+You will need to modify the version and changelog manually in this case. Check
+the existing release PRs for examples on how to do this. Once the branch is
+ready, create a draft PR so that the release team can review it.
+
+The wheel building process and publication to PyPI is handled by the CI.
+Just create a [github release](https://github.com/CQCL/guppylang/releases/new) from the **unmerged** branch.
+The release tag should follow the format used in the previous releases, e.g. `hugr-py-v0.1.1`.
```
Comment (ss2165) [inline DEVELOPMENT.md:160]: mention the need to also merge the changelog/version changes back in to main

**Review by ss2165 — CHANGES_REQUESTED** [review]

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+All the required dependencies should be available. You can automate loading the
+shell by setting up [direnv](https://devenv.sh/automatic-shell-activation/).
+
+### Manual setup
+
+To setup the environment manually you will need:
+
+- Just: https://just.systems/
+- uv `>=0.3`: docs.astral.sh/uv/getting-started/installation
```
[the remark this answers is not in the document]
Comment (aborgna-q) [reply DEVELOPMENT.md]: Formated the links as `[domain](long-url)` for clarity.

**Review by ss2165 — APPROVED** [review]

**Pull request merged**